cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/flow/BasicFlowletContext.java

```diff
@@ -83,6 +86,18 @@ public MetricsCollector load(String key) throws Exception {
         }
       });
 
+    this.producerMetrics = CacheBuilder.newBuilder()
```

Comment:
I think the newer metrics implementation has incorporated the cache (which it wasn't before). I believe we can remove cache usage and just let the metrics api to do so.